Library/ClassMethod.php

```diff
@@ -2441,4 +2475,235 @@ public function areReturnTypesCompatible()
+                    $param = sprintf('Z_PARAM_DOUBLE_OR_NULL(%s, is_null_true)', $name);
+                } else {
+                    $param = sprintf('Z_PARAM_DOUBLE(%s)', $name);
+                }
+
+                break;
+
+            case 'int':
+                if ($hasDefaultNull) {
+                    $param = sprintf('Z_PARAM_LONG_OR_NULL(%s, is_null_true)', $name);
+                } else {
+                    $param = sprintf('Z_PARAM_LONG(%s)', $name);
+                }
+
+                break;
+
+            case 'object':
+                if ($hasDefaultNull) {
+                    $param = sprintf('Z_PARAM_OBJECT_OF_CLASS_OR_NULL(%s)', $name);
+                } else {
+                    $param = sprintf('Z_PARAM_OBJECT(%s)', $name);
+                }
+
+                break;
+
+            case 'resource':
+                if ($hasDefaultNull) {
+                    $param = sprintf('Z_PARAM_RESOURCE_OR_NULL(%s)', $name);
+                } else {
+                    $param = sprintf('Z_PARAM_RESOURCE(%s)', $name);
+                }
+
+                break;
+
+            case 'string':
+                if ($hasDefaultNull) {
+                    $param = sprintf('Z_PARAM_STR_OR_NULL(%s)', $name);
+                } else {
+                    $param = sprintf('Z_PARAM_STR(%s)', $name);
+                }
+
+                break;
+
+            case 'variable':
+                if (isset($parameter['cast']) && $parameter['cast']['type'] === 'variable' && $parameter['cast']['value']) {
+                    $classEntry = $this->detectClassNameEntry($parameter['cast']['value'], $compilationContext);
+                    if ($classEntry !== null) {
+                        if ($hasDefaultNull) {
+                            $param = sprintf('Z_PARAM_OBJECT_OF_CLASS_OR_NULL(%s, %s)', $name, $classEntry);
+                        } else {
+                            $param = sprintf('Z_PARAM_OBJECT_OF_CLASS(%s, %s)', $name, $classEntry);
+                        }
+                    }
+                }
+
+                break;
+        }
+
+        return $param;
+    }
+
+    /**
+     * @param string $className
+     * @param CompilationContext $compilationContext
+     * @return string|null
+     * @throws Exception
+     */
+    private function detectClassNameEntry(string $className, CompilationContext $compilationContext): ?string
+    {
+        if ($this->classDefinition === null) {
+            return null;
+        }
+
+        $isAlias = false;
+        $aliasManager = $this->classDefinition->getAliasManager();
+        if ($aliasManager->isAlias($className)) {
+            $isAlias = true;
+            $className = $aliasManager->getAlias($className);
+        }
+
+        /**
+         * PSR
+         */
+        if (strpos($className, 'Psr') === 0 || strpos($className, '\Psr') === 0) {
+            $className = ltrim($className, '\\');
+
+            $psrHeaderFiles = [
```

Comment:
My proposition - move this psr definition to separate file and use it as constant

Reply:
Yes. This is just for fast tests, will be moved soon.